src/ansys/simai/core/data/model_configuration.py

```diff
@@ -313,11 +314,26 @@ def __get_gc(self):
 
     global_coefficients: list[GlobalCoefficientDefinition] = property(__get_gc, __set_gc)
 
+    def __validate_build_preset(self, val: str):
+        if val not in SupportedBuildPresets:
+            raise InvalidArguments(
+                f"Invalid value for build_preset, build_preset should be one of {list(SupportedBuildPresets.keys())}"
+            )
+
+    def __set_build_preset(self, val: str):
+        self.__validate_build_preset(val)
+        self.__dict__["build_preset"] = val
+
+    def __get_build_preset(self):
+        return self.__dict__.get("build_preset")
+
+    build_preset = property(__get_build_preset, __set_build_preset)
```

Comment:
instead of:
```python
def __validate_build_preset(): ...
def __set_build_preset(): ...
def __get_build_preset(self): ...

build_preset = property(__get_build_preset, __set_build_preset)
[...]
del __set_build_preset, __get_build_preset
```

it looks like we could have just:
```python
def __get_build_preset(self): ...
def __set_build_preset(): ...

build_preset = property(__get_build_preset, __set_build_preset)
```

Reply:
I actually prefer to put these helper functions out of the scope:
1. I like the symantics of it, i.e., since they are deleted methods, it means they are only there to serve a specific purpose
2. the user cannot call them, e.g.,:
```py
bld_conf = ModelConfiguration(
    project=my_project,
    build_preset="debug",
)
```
raises the error:
```sh
AttributeError: 'ModelConfiguration' object has no attribute '__get_build_preset'
```

Reply:
1. thanks pointing out that `__` attributes are hidden from the user. My argument on the semantics is still valid though.
2. although `__validate_build_preset` is not currently used elsewhere, having a dedicate method for it segments better the responsibilities of the methods conceptually.  

Hope this answers everything.